xwiki-platform-core/xwiki-platform-watchlist/xwiki-platform-watchlist-api/src/main/java/org/xwiki/watchlist/internal/RealtimeNotificationGenerator.java

```diff
@@ -115,17 +128,35 @@
     private ConfigurationSource xwikiProperties;
 
     /**
+     * Allow processing of remote events.
+     */
+    private boolean allowRemote;
+
+    /**
      * Default constructor.
      */
     public RealtimeNotificationGenerator()
     {
         super(LISTENER_NAME, EVENTS);
+        allowRemote = false;
+    }
+
+    /**
+     * Component manager initialize class. Called after all dependencies are injected.
+     *
+     * @throws InitializationException if component isn't properly initialized
+     */
+    public void initialize() throws InitializationException
+    {
+        if (TRUE.equals(xwikiProperties.getProperty("watchlist.realtime.allow_remote"))) {
```

Comment:
xwikiProperties.getProperty("watchlist.realtime.allow_remote", false) seems better to me (yes I'm not a fan of how "watchlist.realtime.enabled" is handled), you are redoing its job in a more limited way basically.